x-pack/solutions/security/plugins/security_solution/public/detection_engine/rule_details_ui/pages/rule_details/rule_actions_overflow/index.test.tsx

```diff
@@ -280,4 +305,52 @@ describe('RuleActionsOverflow', () => {
       });
     });
   });
+
+  describe('rule revert to base version flyout', () => {
+    const customizedMockRule = {
+      ...mockRule('id'),
+      rule_source: { type: 'external', is_customized: true } as ExternalRuleSource,
+    };
+    test('it shows the revert action when rule is prebuilt and customized', () => {
+      const { getByTestId } = render(
+        <RuleActionsOverflow
+          showBulkDuplicateExceptionsConfirmation={showBulkDuplicateExceptionsConfirmation}
+          showManualRuleRunConfirmation={showManualRuleRunConfirmation}
+          rule={customizedMockRule}
+          userHasPermissions
+          canDuplicateRuleWithActions={true}
+          confirmDeletion={() => Promise.resolve(true)}
+          openRuleDiffFlyout={openRuleDiffFlyoutMock}
+          isRevertBaseVersionDisabled={false}
+        />,
+        { wrapper: TestProviders }
+      );
+      fireEvent.click(getByTestId('rules-details-popover-button-icon'));
+      expect(getByTestId('rules-details-menu-panel')).toHaveTextContent(
+        'Revert to Elastic version'
+      );
+
+      fireEvent.click(getByTestId('rules-details-revert-rule'));
```

Comment:
nit: It's possible to explicitly find an element by text so these two lines could be replaced with

```ts
const revertRuleButton = within(getByTestId('rules-details-menu-panel')).getByText('Revert to Elastic version');

fireEvent.click(revertRuleButton);
```